.github/workflows/docs-ci.yaml

```diff
@@ -0,0 +1,53 @@
+name: documentation CI
+
+on:
+  workflow_call:
+    inputs:
+      git_tag:
+        type: string
+        description: The git tag (version) from the calling workflow
+        required: true
+
+jobs:
+  docs-ci:
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GH_TOKEN_NVIDIA_CI_CD }}
+      TAG: ${{ inputs.git_tag || 'latest' }}
+      PR_TITLE_PREFIX: "task: update documentation for"
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        token: ${{ secrets.GH_TOKEN_NVIDIA_CI_CD }}  # token must be explicitly set here for push to work in following step
+    - name: Make docs
+      run: |
+        make helm-docs api-docs generate-docs-versions-var doca-driver-build-update
+    - name: Close any existing documentation PRs
+      run: |
+        for pr_number in $(gh pr list --search "$PR_TITLE_PREFIX" --json number --jq ".[].number"); do
+          gh pr close $pr_number
+        done
+    - name: Create PR
+      env:
+        DOWNSTREAM_REPO_OWNER: nvidia-ci-cd
+        DOWNSTREAM_FEATURE_BRANCH: update-docs-for-${{ env.TAG }}
+        UPSTREAM_REPO_OWNER: Mellanox
+        UPSTREAM_DEFAULT_BRANCH: main
+        COMMIT_MESSAGE: ${{ env.PR_TITLE_PREFIX }} ${{ env.TAG }}
+      run: |
+        git config user.name  nvidia-ci-cd
+        git config user.email svc-cloud-orch-gh@nvidia.com
+        gh repo fork --remote --default-branch-only
+        gh repo sync $DOWNSTREAM_REPO_OWNER/${{ github.event.repository.name }} --source $UPSTREAM_REPO_OWNER/${{ github.event.repository.name }} --branch $UPSTREAM_DEFAULT_BRANCH
+
+        git checkout -b $DOWNSTREAM_FEATURE_BRANCH
+        git status
+        git add docs
```

Comment:
I believe https://github.com/peter-evans/create-pull-request would handle the PR creation/cleanup. So you would only make sure that the docs are up to date.

Reply:
For now, if the current implementation works (and doesn't rely on 3rd party code), I believe we should stick with it and move forward.

But I'll be sure to look into this 3rd party action for next time I need something like this. Thanks for sharing with me!

Reply:
Ok, that's not a blocker. But I believe it would be nice, as it does not open/close multiple PRs but keep updating a single PR as long as possible.